File: Recursion/zigZag.java
import java.util.Scanner;
public class zigZag
{
    public static Scanner scn = new Scanner(System.in);
    public static void pzz(int n){
        if(n == 0)
        {
            return;
        }
        System.out.print(n + " ");
        pzz(n - 1);
        System.out.print(n + " ");
        pzz(n - 1);
        System.out.print(n + " ");
    }

    public static void main(String[] args)
    {
        int n = scn.nextInt();
        pzz(n);
    }
}

File: Recursion/zigZag.cpp
#include<iostream>
using namespace std;

void pzz(int n)
{
    if(n == 0)
    {
        return;
    }
    cout << n << " ";
    pzz(n - 1);
    cout << n << " ";
    pzz(n - 1);
    cout << n << " ";
}



int main()
{
    int n;
    cin>>n;
    pzz(n);
    return 0;
}